frontend/src/app/workflows/workflows.component.ts

```diff
@@ -0,0 +1,166 @@
+import {IOutputData, SplitComponent} from 'angular-split';
+import {workflowsSchema} from './model/helper/workflows.schema';
+import {environment} from '../../environments/environment';
+import {LintService} from '../shared/lint.service';
+
+@Component({
+  selector: 'app-workflows',
+  templateUrl: './workflows.component.html',
+  styleUrls: ['./workflows.component.scss']
+})
+export class WorkflowsComponent implements OnInit {
+  @ViewChild('split') split!: SplitComponent;
+
+  public content!: any;
+  public codemirrorOptions: any;
+
+  public generateResult!: GenerateResult;
+
+  public currentExampleDesc: string = 'Select example';
+  public examplesList = ['New Workflow', 'Data Modelling', 'Microservices', 'Pages'];
+
+  public showIframeHider = false;
+  public newPageIndex: number = 0;
+  public currentDisplay: 'pages' | 'objects' | 'class' = 'pages';
+
+  private ajv!: Ajv;
+  private validate!: ValidateFunction;
+  public loading: boolean = false;
+
+  constructor(
+    private toastService: ToastService,
+    private zone: NgZone,
+    private fulibWorkflowsService: WorkflowsService,
+    private privacyService: PrivacyService,
+    private http: HttpClient,
+    private lintService: LintService,
+  ) {
+    // https://angular.io/api/core/NgZone
+    const generateHandler = () => this.zone.run(() => this.generate());
+
+    this.codemirrorOptions = {
+      lineNumbers: true,
+      mode: 'yaml',
+      extraKeys: {
+        'Ctrl-Space': 'autocomplete',
+        'Ctrl-S': generateHandler,
+        'Cmd-S': generateHandler,
+      },
+      autofocus: true,
+      tabSize: 2,
+    };
+  }
+
+  ngOnInit() {
+    this.setExample(this.examplesList[0]);
+    this.ajv = new Ajv();
+    this.validate = this.ajv.compile(workflowsSchema);
+  }
+
+  changeExampleContent(index: number) {
+    this.currentExampleDesc = this.examplesList[index];
+    this.setExample(this.currentExampleDesc);
+  }
+
+  generate() {
+    this.privacyService.setStorage('workflows', this.content);
+
+    if (this.loading) return;
+
+    if (!this.content.includes('- workflow: ')) {
+      this.toastService.error('Lint Error', 'Needs at least one workflow note (best at the beginning)');
+      return;
+    }
+
+    // Replace tabs with two spaces for js-yaml and snakeyaml parser
+    this.content = this.content.replace(/\t/g, '  ');
+
+    const validYaml = this.lintService.lintYamlString(this.content, this.validate);
+
+    if (!validYaml) {
+      const errorMessage = this.lintService.evaluateErrorMessage(this.validate);
+      this.toastService.error('Lint Error', errorMessage);
+      return;
+    }
+
+    this.loading = true;
+
+    this.fulibWorkflowsService.generate(this.content).subscribe(
+      (answer) => {
+        this.generateResult = answer
+        this.loading = false;
+      },
+      (error: any) => {
+        this.loading = false;
+
+        let errorMessage = error.error.status + '\n';
+        errorMessage += error.error.message;
```

Comment:
Could use template string and const (or no variable at all)

Reply:
Check again